backend/pkg/api/data_access/general.go

```diff
@@ -46,3 +48,55 @@ func (d *DataAccessService) GetNamesAndEnsForAddresses(ctx context.Context, addr
 	}
 	return nil
 }
+
+// helper function to sort and apply pagination to a query
+// 1st param is the list of all columns necessary to sort the table deterministically; it defines their precedence and sort direction
+// 2nd param is the requested sort column; it may or may not be part of the default columns
+func applySortAndPagination(defaultColumns []types.SortColumn, primary types.SortColumn, cursor types.GenericCursor) ([]exp.OrderedExpression, exp.Expression) {
```

Comment:
Yeah thought this would help us going forward, glad you like it! I'm also planning to refactor some queries with this.

Feel free to adjust/extend if useful :+1: